src/blockdev.rs

```diff
@@ -519,6 +520,36 @@ pub struct SavedPartitions {
     partitions: Vec<(u32, GPTPartitionEntry)>,
 }
 
+fn translate_mbr_types_to_gpt(sys: u8) -> Uuid {
+    // non inclusive best effort mapping of MBR types to GPT types using the below references
+    // https://en.wikipedia.org/wiki/GUID_Partition_Table#Partition_type_GUIDs && https://tldp.org/HOWTO/Partition-Mass-Storage-Definitions-Naming-HOWTO/x190.html
+
+    match sys {
+        // Linux filesystem
+        0x01 => uuid!("0FC63DAF-8483-4772-8E79-3D69D8477DE4"),
+        0x04 => uuid!("0FC63DAF-8483-4772-8E79-3D69D8477DE4"),
+        0x06 => uuid!("0FC63DAF-8483-4772-8E79-3D69D8477DE4"),
+        // Linux Filesystem Data
+        0x83 => uuid!("0FC63DAF-8483-4772-8E79-3D69D8477DE4"),
+        // Linux LVM
+        0x8e => uuid!("E6D6D379-F507-44C2-A23C-238F2A3DF928"),
+        // Linux Swap
+        0x82 => uuid!("0657FD6D-A4AB-43C4-84E5-0933C84B4F4F"),
+        _ => uuid!("00000000-0000-0000-0000-000000000000"),
+    }
+}
+
+fn translate_mbr_to_gpt(mbr_partition: &MBRPartitionEntry) -> GPTPartitionEntry {
+    let mut gpt_partition = GPTPartitionEntry::empty();
+
+    gpt_partition.partition_type_guid = *translate_mbr_types_to_gpt(mbr_partition.sys).as_bytes();
+    gpt_partition.unique_partition_guid = *Uuid::new_v4().as_bytes();
+    gpt_partition.starting_lba = u64::from(mbr_partition.starting_lba);
```

Comment:
I think it's a bit more common to just use `as u64` here.  That said there's also the `try_from()` variant which might make sense to actually validate things.